src/AuthenticationResponse.ts

```diff
@@ -39,6 +41,16 @@ export default class AuthenticationResponse {
     verifiedJwt: SIOP.VerifiedAuthenticationRequestWithJWT,
     responseOpts: SIOP.AuthenticationResponseOpts
   ): Promise<SIOP.AuthenticationResponseWithJWT> {
+    const pejs: PEJS = new PEJS();
+    // bunch of ifs to make sure we need to call evaluate
+    const pd: PresentationDefinition =
+      verifiedJwt.payload.claims['id_token']['verifiable_presentations'].presentation_definition;
+    const result: EvaluationResults = pejs.evaluate(pd, responseOpts.vp);
+    if (result.errors.length) {
+      throw new Error(`${SIOPErrors.EVALUATE_PRSENTATION_EXCHANGE_FAILED}`);
+    }
+    const ps = pejs.submissionFrom(pd, responseOpts.vp.getVerifiableCredentials());
+    responseOpts.vp.getRoot()['presentation_submission'] = ps;
```

Comment:
Hmzz, the options are being passed in by the user, I don't like we start changing the supplied options. So we need to explode the options into a new options object or something